ui/src/screens/Posts/hooks/usePostsReducer.ts

```diff
@@ -85,47 +106,53 @@ function PostsReducer(state: PostsState, action: PostsActionTypes): PostsState {
 // Hook
 
 interface PostsActions {
-  setActiveJob(id: string): void;
+  setActivePost(id: string, category: CategoryTypes): void;
   toggleFavouriteJob(job: PostInterface): void;
   updateJobs(jobs: PostInterface[]): void;
 }
 
-interface JobsResponseData {
-  jobs: PostInterface[];
-}
+// interface PostsResponseData {
+//   jobs?: PostInterface[];
+//   talents?: PostInterface[];
+// }
 
 export default function usePostsReducer(): [PostsState, PostsActions] {
   const [state, dispatch] = React.useReducer(PostsReducer, initialState);
 
-  const setActiveJob = React.useCallback((id: string): void => {
-    dispatch({ type: SET_ACTIVE_JOB, payload: id });
+  const setActivePost = React.useCallback((id: string, category: CategoryTypes): void => {
+    dispatch({ type: SET_ACTIVE_POST, payload: { id, category } });
   }, []);
   const toggleFavouriteJob = React.useCallback((job: PostInterface): void => {
     dispatch({ type: TOGGLE_FAVOURITE_JOB, payload: job });
   }, []);
   const updateJobs = React.useCallback((jobs: PostInterface[]): void => {
     dispatch({ type: UPDATE_JOBS, payload: jobs });
   }, []);
+  const updateTalents = React.useCallback((talents: PostInterface[]): void => {
+    dispatch({ type: UPDATE_TALENTS, payload: talents });
+  }, []);
   const actions: PostsActions = React.useMemo(() => {
     return {
-      setActiveJob,
+      setActivePost,
       toggleFavouriteJob,
       updateJobs,
+      updateTalents,
     };
-  }, [setActiveJob, toggleFavouriteJob, updateJobs]);
+  }, [setActivePost, toggleFavouriteJob, updateJobs, updateTalents]);
 
-  const match = useRouteMatch<{ id: string }>("/posts/:id");
-  const id = match?.params?.id;
+  const { category: categoryString, id } = useParams<{ category: string; id: string }>();
+  const category = categoryString as CategoryTypes;
 
   React.useEffect(() => {
-    axios.get<JobsResponseData>("/api/jobs").then((res) => {
-      updateJobs(res.data.jobs);
-      if (id) setActiveJob(id);
+    axios.get(`/api/${category}`).then((res) => {
```

Comment:
@lxcid I had some trouble with this `axios.get` part.

You earlier added the jobs response interface: `axios.get<JobsResponseData>(...)`

But now, depending on category, the response will either be
`{ jobs: PostInterface[] }`
or
`{ talents: PostInterface[] }`

I can't figure out how to do this, or if it even makes sense, so I removed it for now.

Reply:
noted! I think I can figure that out later. :)